Tests/lib/taskRunner.ts

```diff
@@ -107,11 +106,13 @@ export class TaskRunner extends events.EventEmitter {
 			shell.cp('-R', this._taskSrcPath, this._tempPath);
 		}
 
-		// delete it's linked copy of vsts-task-lib so it uses the mocked task-lib above
-		var taskLibPath = path.join(this._taskPath, 'node_modules', 'vsts-task-lib');
-		if (shell.test('-d', taskLibPath)) {
-			shell.rm('-rf', taskLibPath);
-		}
+		// delete all nested copies of vsts-task-lib so the mock vsts-task-lib is used (copied above)
+		shell.find(this._taskPath)
+			.forEach(function (item) {
+				if (path.basename(item) == 'vsts-task-lib' && path.basename(path.dirname(item)) == 'node_modules') {
+					shell.rm('-rf', item);
+				}
+			});
```

Comment:
After I fixed the specific gradle test issue that was failing because of commandFromString, issues started manifested in other tests. It took an extremely long time to figure out what was going on.

1) some _suite.ts files load task scripts directly directly (i.e. not via taskRunner) and fail if the mock vsts-task-lib is used instead. so we can't delete every vsts-task-lib folder under _test/Tasks (I removed that logic from make.js).

on a side note, this probably why we see .taskkey files after running gulp test.

2) npm install wasn't run for Common folders in the old gulp process. After I made the change not to delete vsts-task-lib under _test/Tasks, then other problems started manifested during taskRunner runs. If a task loaded a common module at the start of its entry script, then the common module would load the real vsts-task-lib which would yank inputs out of the ENV. Then when the script called getInput(foo, require) => fail.

The fix is to change taskRunner to delete all nested vsts-task-lib directories.